### PR TITLE
BeatSaberModManager: init at 0.0.2

### PR DESCRIPTION
**Pull request by Atemu**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/BeatSaberModManager/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    sha256 = "sha256-6+9pWr8jJzs430Ai2ddh/2DK3C2bQA1e1+BNDrKhyzY=";
+    fetchSubmodules = true; # It vendors BSIPA-Linux
+  };
+
+  nugetDeps = ./deps.nix;
+
+  dotnetInstallFlags = [
+    # FIXME It doesn't like that for some reason beyond my ability to package dotnet
+    "-p:PublishSingleFile=false"
+  ];
```
Comment (Atemu) [inline pkgs/games/BeatSaberModManager/default.nix:30]: Specifically, it fails with

```
Executing dotnetInstallHook
MSBuild version 17.3.1+2badb37d1 for .NET
  IPA-lib -> /nix/store/hsx50dc5yly2bqinqcdr9sz6xs9zfawc-BeatSaberModManager-0.0.2/lib/BeatSaberModManager/
/nix/store/a5x48hb982y79kkh39n8wh16b62vi1iz-dotnet-sdk-6.0.401/sdk/6.0.401/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(102,5): error NETSDK1097: It is not supported to publish an application to a single-file without specifying a RuntimeIdentifier. You must either specify a RuntimeIdentifier or set PublishSingleFile to false. [/build/source/BeatSaberModManager/BeatSaberModManager.csproj]
```

If anyone more knowledgable in dotnet knows how to fix this without disabling single-file, let me know.

Comment (Atemu) [reply pkgs/games/BeatSaberModManager/default.nix]: Found a solution but it's not pretty.

**Comment by Atemu** [comment]
I'd love to see this merged before branch-off.

File: pkgs/games/BeatSaberModManager/default.nix (hunk 1 above)
Comment (Atemu) [reply pkgs/games/BeatSaberModManager/default.nix]: Created an issue https://github.com/affederaffe/BeatSaberModManager/issues/5

**Review by K900 — APPROVED** [review]
Diff generally LGTM, but I don't have a headset to test on anymore.

**Comment by Atemu** [comment]
@K900 thanks!

You don't need a headset btw; you can observe all of BSMM's effects with just a BeatSaber installation. It's just a fancy GUI to download files and shuffle them around. 

Speaking of 'BSMM', do you think adding an alias would be good?

**Comment by Atemu** [comment]
Thanks!

**Pull request merged**